### PR TITLE
Add Memgraph support

### PR DESCRIPTION
**Pull request by katarinasupe**
I added Memgraph support by extending `Neo4jGraph` class. In order to do that, I had to mark `schema` and `structuredSchema` as `protected` instead of `private`, so I was wondering if that's okay. Other than that, I mostly followed what we did in LangChain Python and I plan to fix a couple of things there to better align with this integration. 

@tomasonjo we spoke recently about Python integration so tagged you for awareness here.

I plan to add docs and example, but first wanted to see if this is okay to start with and if I have to first fix something in this contribution.

### CODE REVIEW

File: libs/langchain-community/src/graphs/memgraph_graph.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  }
+
+  static async initialize(config: MemgraphGraphConfig): Promise<MemgraphGraph> {
+    const graph = new MemgraphGraph(config);
+
+    try {
+      await graph.verifyConnectivity();
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      console.log("Failed to verify connection.");
```
Comment (bracesproul) [inline libs/langchain-community/src/graphs/memgraph_graph.ts:47]: nit
```suggestion
      console.error("Failed to verify connection.");
```

File: libs/langchain-community/src/graphs/memgraph_graph.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    try {
+      await graph.refreshSchema();
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      throw new Error(error.message);
+    } finally {
+      console.log("Schema refreshed successfully.");
```
Comment (bracesproul) [inline libs/langchain-community/src/graphs/memgraph_graph.ts:56]: Not sure we want this logging every time?
```suggestion
      console.debug("Schema refreshed successfully.");
```

File: libs/langchain-community/src/graphs/memgraph_graph.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      const rawSchema = rawSchemaQueryResult?.[0]?.schema;
+
+      this.structuredSchema = {
+        nodeProps: rawSchema.node_props,
+        relProps: rawSchema.rel_props,
+        relationships: rawSchema.relationships,
+      };
+
+      // Format node properties
+      const formattedNodeProps = Object.entries(rawSchema.node_props)
```
Comment (bracesproul) [inline libs/langchain-community/src/graphs/memgraph_graph.ts:74]: Why are we using `rawSchema` here instead of `this.structuredSchema`?

Comment (bracesproul) [reply libs/langchain-community/src/graphs/memgraph_graph.ts:74]: Same for all occurrences below

File: libs/langchain-community/src/graphs/neo4j_graph.ts [hunk 4]
```diff
@@ -42,9 +42,9 @@ export class Neo4jGraph {
 
   private database: string;
 
-  private schema = "";
+  protected schema = "";
 
-  private structuredSchema: StructuredSchema = {
+  protected structuredSchema: StructuredSchema = {
```
Comment (bracesproul) [inline libs/langchain-community/src/graphs/neo4j_graph.ts:47]: 👍

File: libs/langchain-community/src/graphs/tests/memgraph_graph.int.test.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    // expect(graph.getSchema()).toMatchInlineSnapshot(`
+    //   "Node properties are the following:
+    //   Node name: 'Actor', Node properties: [{"property":"name","type":"str"}]
+    //   Node name: 'Movie', Node properties: [{"property":"title","type":"str"}]
+    //   Relationship properties are the following:
+    //   Relationship name: 'ACTED_IN', Relationship properties: [{"property":"roles","type":"tuple"}]
+    //   The relationships are the following:
+    //   (:Actor)-[:ACTED_IN]->(:Movie)"
+    // `);
```
Comment (bracesproul) [inline libs/langchain-community/src/graphs/tests/memgraph_graph.int.test.ts:43]: Should this be uncommented or deleted?

**Review by bracesproul — COMMENTED** [review]
Left a couple comments, but overall this looks good! Also, it would be great to include docs in this PR, makes it easier to review if it's all in one.
Thanks for this!

**Comment by katarinasupe** [comment]
Hi @bracesproul, thank you for the review. I will update the PR in a couple of days when I am back from the trip 😄

File: libs/langchain-community/src/graphs/memgraph_graph.ts (hunk 3 above)
Comment (katarinasupe) [reply libs/langchain-community/src/graphs/memgraph_graph.ts:74]: It's the same if I use `rawSchema.node_props` or `this.structuredSchema.nodeProps` (similar for other occurrences). Do you prefer the latter, and if yes, why? I am not a ts/js expert, so any advice is more than welcome 😄

File: libs/langchain-community/src/graphs/tests/memgraph_graph.int.test.ts (hunk 5 above)
Comment (katarinasupe) [reply libs/langchain-community/src/graphs/tests/memgraph_graph.int.test.ts:43]: I will uncomment it :)

**Review by bracesproul — APPROVED** [review]
LGTM! Thank you for adding this & docs!

**Pull request merged**